src/main/java/com/xatkit/core/recognition/processor/PerspectiveapiInterface.java

```diff
@@ -0,0 +1,281 @@
+     * @param doNotStore       if true then data will not be stored to PerspectiveAPI. Default is false
+     * @param clientToken      the client token
+     * @param sessionId        the session id
+     * @param attributesPrefix the attributes prefix for the keys of the map they will be stored in
+     */
+    PerspectiveapiInterface(@NonNull String apiKey, String language, Boolean doNotStore, String clientToken,
+                            String sessionId, String attributesPrefix) {
+
+        this.apiKey = apiKey;
+        this.language = language != null ? language : "en";
+        this.doNotStore = doNotStore != null ? doNotStore : false;
+        this.clientToken = clientToken;
+        this.sessionId = sessionId;
+        this.attributesPrefix = attributesPrefix;
+        languageAttributes = new HashMap<>();
+        languageAttributes.put("en", new AttributeType[] {
+                AttributeType.TOXICITY,
+                AttributeType.SEVERE_TOXICITY,
+                AttributeType.TOXICITY_FAST,
+                AttributeType.IDENTITY_ATTACK,
+                AttributeType.INSULT,
+                AttributeType.PROFANITY,
+                AttributeType.THREAT,
+                AttributeType.SEXUALLY_EXPLICIT,
+                AttributeType.FLIRTATION,
+                AttributeType.ATTACK_ON_AUTHOR,
+                AttributeType.ATTACK_ON_COMMENTER,
+                AttributeType.INCOHERENT,
+                AttributeType.INFLAMMATORY,
+                AttributeType.LIKELY_TO_REJECT,
+                AttributeType.OBSCENE,
+                AttributeType.SPAM,
+                AttributeType.UNSUBSTANTIAL
+        });
+        languageAttributes.put("es", new AttributeType[] {
+                AttributeType.TOXICITY,
+                AttributeType.SEVERE_TOXICITY,
+                AttributeType.IDENTITY_ATTACK_EXPERIMENTAL,
+                AttributeType.INSULT_EXPERIMENTAL,
+                AttributeType.PROFANITY_EXPERIMENTAL,
+                AttributeType.THREAT_EXPERIMENTAL
+        });
+        this.setRequestedAttributes();
+    }
+
+    /**
+     * Sets the attributes available for the chosen language
+     */
+    private AttributeType[] setRequestedAttributes() {
+        if (language.equals("es")) {
+            requestedAttributes = languageAttributes.get("es");
+        }
+        else {
+            requestedAttributes = languageAttributes.get("en");
+        }
+        return requestedAttributes;
+    }
+
+    /**
+     * Sets the comment that will be analyzed
+     */
+    private String setCommentText(String commentText) {
+        this.commentText = commentText;
+        return commentText;
+    }
+
+    /**
+     * Analyzes the comment and returns the punctuation for each attribute
+     *
+     * @param comment the comment
+     * @return the hash map with attributes and their respective scores
+     */
+    public HashMap<String, Double> analyzeRequest(String comment) {
```

Comment:
`HashMap` is too specific: client code does not need to know what kind of map we are using. More information in [this presentation by Mario Fusco](https://fr.slideshare.net/mariofusco/lets-make-a-contract-the-art-of-designing-a-java-api-231818668/33)